packages/Interface/src/DTK_C_API.cpp

```diff
@@ -349,98 +363,135 @@ bool DTK_is_initialized() { return DataTransferKit::isInitialized(); }
 
 void DTK_finalize() { DataTransferKit::finalize(); }
 
-void DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
-                       void ( *f )(), void *user_data )
+int DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
```

Comment:
I thought the point of using errno was that we wouldn't return error codes? (i.e. the function signatures would not change)